### PR TITLE
NC-354 Refactor tests

### PR DESCRIPTION
**Pull request by pgan002**

### CODE REVIEW

File: tests/test_date.py [hunk 1]
```diff
@@ -34,6 +54,215 @@ def test_time_formats(self):
...
+
+        # Future weekday
+        self.assert_date('wed', datetime(2017, 6, 21))
+        self.assert_date('last wed', datetime(2017, 6, 14))
+        self.assert_date('previous wed', datetime(2017, 6, 14))
+        self.assert_date('next wed', datetime(2017, 6, 21))
+        self.assert_date('upcoming wed', datetime(2017, 6, 21))
+
+        # Current weekday
+        self.assert_date('fri', datetime(2017, 6, 16))
```
Comment (justonj) [inline tests/test_date.py]: Should be next friday datetime(2017, 6, 23)

File: tests/test_date.py [hunk 2]
```diff
@@ -34,6 +54,215 @@ def test_time_formats(self):
...
+        self.assert_date('this fri', datetime(2017, 6, 16))
```
Comment (justonj) [inline tests/test_date.py]: Should be next friday datetime(2017, 6, 23)

**Review by justonj — CHANGES_REQUESTED** [review]

File: tests/test_date.py (hunk 1 above)
Comment (pgan002) [reply tests/test_date.py]: That may be fine for use cases you have in mind, and can be achieved by specifying parameter `context=CONTEXT_NEXT`. But I don't think it is a good default of a general-purpose library.

File: tests/test_date.py (hunk 2 above)
Comment (pgan002) [reply tests/test_date.py]: As above.

File: tests/test_date.py [hunk 3]
```diff
@@ -34,6 +54,215 @@ def test_time_formats(self):
             # TODO: 11pm, dec 31
             # TODO: dec 31 @ 11pm
 
+    def test_date_and_time(self):
+        now = datetime.now()
+
+        date = Date('01/10/2015 at 7:30pm')
```
Comment (dpathania1) [inline tests/test_date.py]: we can use for loop here as well to avoid replicating the assert statements.

File: tests/test_date.py [hunk 4]
```diff
@@ -34,6 +54,215 @@ def test_time_formats(self):
...
+        for param in ts, str(ts):
+            date = Date(param)
+            self.assertEqual(date.year, 2013)
+            self.assertEqual(date.month, 7)
+            self.assertEqual(date.day, 24)
+
+        # offset timezones
+        self.assertEqual(Date('2014-03-06 15:33:43.764419-05').hour, 20)
+
+        for x in ['yestserday', 'Satruday', Exception]:
```
Comment (dpathania1) [inline tests/test_date.py]: This can be a separate test.

File: tests/test_date.py [hunk 5]
```diff
@@ -34,6 +54,215 @@ def test_time_formats(self):
...
+        self.assertEqual(d.month, 4)
+        self.assertEqual(d.day, 15)
+        self.assertEqual(d.hour, 5)
+        self.assertEqual(d.minute, 40)
+        self.assertEqual(d.second, 14)
+        self.assertEqual(str(d.date), '2013-04-15 05:40:14.010001')
+
+    def test_next_prev(self):
+        # Past month
+        self.assert_date('nov', datetime(2017, 11, 1))
```
Comment (dpathania1) [inline tests/test_date.py]: isnt nov future month?

Comment (pgan002) [reply tests/test_date.py]: @dpathania1 thank you for the detailed review!

**Review by justonj — APPROVED** [review]

**Pull request merged**